activerecord/lib/active_record/connection_adapters/abstract/transaction.rb

```diff
@@ -109,14 +109,22 @@ def dirty?
         @dirty
       end
 
-      def add_record(record, ensure_finalize = true)
+      # Add the record to the current transaction, used by after_commit (etc) callbacks.
+      # Returns true if the record was added to the transaction, false if it had been previously added.
+      def uniquely_add_record(record, ensure_finalize = true)
```

Comment:
I don't need to rename this method if it will cause problems, but as far as I can tell it's private API. And I think renaming makes it a bit more clear why it's returning a bool.